src/lib.rs

```diff
@@ -181,7 +184,7 @@ pub struct RestrictedMemory<M: Memory> {
 
 impl<M: Memory> RestrictedMemory<M> {
     pub fn new(memory: M, page_range: core::ops::Range<u64>) -> Self {
-        assert!(page_range.end < u64::MAX / WASM_PAGE_SIZE);
+        assert!(page_range.end <= MAX_PAGES);
```

Comment:
Is it intentional you include the equality?

Reply:
Yes, because the right bound in Range is excluded.